src/rock/models/finances/models/Transaction.js

```diff
@@ -209,6 +209,10 @@ export default class Transaction extends Rock {
           }),
         { cache },
       )
+      .then((x) => {
+        if (!limit) return x;
+        return x.slice(offset, limit + offset);
```

Comment:
I think this change needs to be reverted as the `TransactionTable` could eventually return a very large array that JS may not be able to handle. If `limit` is the issue, then we should add an `if` statement before the query instead

Reply:
@richarddubay locally returning this to use `limit` and `offset` isn't causing any issues for me (I've added some extra protection too just in case by defaulting limit and offset to 0). Do you see the same? Or do you still have issues after reverting this slice of code?

Reply:
@Lepozepo If I comment out the `.then...` section and return to passing `limit` and `offset` to Sequelize, then giving history breaks for me. It just says that I have no history at all.